skopt/space/space.py

```diff
@@ -69,6 +69,12 @@ def check_dimension(dimension, transform=None):
     if not isinstance(dimension, (list, tuple, np.ndarray)):
         raise ValueError("Dimension has to be a list or tuple.")
 
+    # Dimension description with single value is assumed to be used
+    # in order to fix dimension value. Is useful for BayesSearchCV,
+    # when you for example want to fix what model is used in Pipeline.
```

Comment:
How about: "A `Dimension` described by a single value is assumed to be a `Categorical` dimension. This can be used in `BayesSearchCV` to define subspaces that fix one value, e.g. to choose the model type."

Though we should create a user facing documentation/example somewhere explaining this as well. It is a pretty cool feature, but you need to be thinking the right way to "get it". At least I need to think about it a bit each time ;)

Reply:
Yup +1 
There is an example actually in the notebook, will add reference in comment